plugins/analysis/java-example/build.gradle.kts

```diff
@@ -0,0 +1,29 @@
+plugins {
+  `java-library`
+}
+
+java {
+  toolchain {
+    languageVersion.set(JavaLanguageVersion.of(11))
+  }
+}
+
+dependencies {
+  implementation(libs.kotlin.stdlibCommon)
+  implementation(libs.kotlin.stdlibJDK8)
+  implementation(projects.arrowAnalysisTypes)
+  // compileOnly(projects.arrowAnalysisLaws)
+  compileOnly(projects.arrowAnalysisJavaPlugin)
+  compileOnly(projects.arrowAnalysisCommon)
+  annotationProcessor(projects.arrowAnalysisJavaPlugin)
+}
+
+tasks.withType<JavaCompile> {
+  javaCompiler.set(javaToolchains.compilerFor {
+    languageVersion.set(JavaLanguageVersion.of(11))
+  })
+  options.compilerArgs.addAll(listOf(
+    "-parameters",
+    "-Xplugin:ArrowAnalysisJavaPlugin"
```

Comment:
We may have to also have this in the Gradle plugin in addition to allowing this kind of usage. the Gradle plugin can apply these settings automatically over Java projects.

Reply:
Indeed, that is the next step. It needs some further investigation, since we cannot reuse the Arrow Meta common implementation, as it targets Kotlin projects.